### PR TITLE
release: 10.0.0

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes https://github.com/rome/tools/issues/3319
Closes https://github.com/rome/tools/issues/3571

## Test Plan

[Rendered website](https://deploy-preview-3526--rometools.netlify.app/).
[Rendered `CHANGELOG.md`](https://github.com/rome/tools/blob/release/10.0.0/CHANGELOG.md).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/01_bug.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ body:
     id: environment
     attributes:
       label: Environment information
-      description: Please share any information about your environment that can help debug your issue (OS and OS version, CPU architecture, installation method, etc.).
-      render: bash
+      description: Run the command `rome range` and paste its output here. Please review it, in case there are sensitive information you don't want to share.
```
Comment (nstepien) [inline .github/ISSUE_TEMPLATE/01_bug.yml:18]: typo?
```suggestion
      description: Run the command `rome rage` and paste its output here. Please review it, in case there are sensitive information you don't want to share.
```

File: editors/vscode/README.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
 
 Adds support for the Rome language server to provide formatting, diagnostics, and code actions.
 
+## How the extension works
+
+The extension acts as a language server, which means that you will get all the features
+provided by the extension (formatting, diagnostics, etc.) for any file supported by Rome.
+
+The extension will also try to discover the `rome.json` file, which needs to be placed
+at the root of your workspace.
```
Comment (MichaReiser) [inline editors/vscode/README.md:11]: The extension automatically discovers the `rome.json` file in the workspace's root directory.

Comment (MichaReiser) [reply editors/vscode/README.md:11]: We should add a section explaining how to make the formatter our default formatter or link to the documentation

## Getting started

...

File: editors/vscode/README.md [hunk 3]
```diff
@@ -2,6 +2,28 @@
...
+
+## Known limitations
+
+There's a number of known limitations that the team plans to remove in the following releases.
```
Comment (MichaReiser) [inline editors/vscode/README.md:15]:
```suggestion
There are a few limitations that the team plans to remove in the following releases.
```

File: editors/vscode/README.md [hunk 4]
```diff
@@ -2,6 +2,28 @@
...
+
+- at the moment the discovery of `rome.json` is only limited to the root of the workspace;
```
Comment (MichaReiser) [inline editors/vscode/README.md:17]:
```suggestion
- the discovery of the `rome.json` is limited to the root of the workspace
```

File: editors/vscode/README.md [hunk 5]
```diff
@@ -2,6 +2,28 @@
...
+- changes to the configuration `rome.json` won't be picked automatically;
```
Comment (MichaReiser) [inline editors/vscode/README.md]:
```suggestion
- You need to restart VS Code or Rome's language server for configuration changes in the `rome.json` to take effect.
```

I still hope that some one us has time to fix this by handling the configuration changes in the VS code extension

File: editors/vscode/README.md [hunk 6]
```diff
@@ -2,6 +2,28 @@
...
+The extension will also try to discover the `rome.json` file, which needs to be placed
+at the root of your workspace.
+
+## Known limitations
+
+There's a number of known limitations that the team plans to remove in the following releases.
+
+- at the moment the discovery of `rome.json` is only limited to the root of the workspace;
+- changes to the configuration `rome.json` won't be picked automatically;  
+- when updating `rome` inside you `package.json`, the rome server needs to be restarted;
```
Comment (MichaReiser) [inline editors/vscode/README.md]:
```suggestion
- You must restart VS Code after upgrading `rome` to a new version.
```

File: editors/vscode/README.md [hunk 7]
```diff
@@ -2,6 +2,28 @@
...
+- if you don't want Rome's diagnostics in projects that don't use Rome, you need to disable
+the extension
```
Comment (MichaReiser) [inline editors/vscode/README.md]: - The Rome extension runs on all projects, even if they don't have a `rome.json` file. That's why you must disable the Rome extensions for workspaces where you don't want Rome enabled.

File: npm/rome/README.md [hunk 8]
```diff
@@ -45,18 +45,32 @@ and top-notch error recovery.
...
-npm i rome@next
+npm install --save-dev rome 
 ```
 
 ## Usage
 
+Start by creating an **optional** configuration file:
+
+```shell
+npx rome init
```
Comment (MichaReiser) [inline npm/rome/README.md:56]: We should either use `npx` everwhere or only use `rome` (I think this is preferred)

File: npm/rome/README.md [hunk 9]
```diff
@@ -45,18 +45,32 @@ and top-notch error recovery.
...
-For complete documentation, please visit the [official website].
+Run the linter:
+
+```shell
+rome check ./path ./path/to/file.js
+```
+
+## Documentation
+
+For a complete documentation, please visit the [official website].
```
Comment (MichaReiser) [inline npm/rome/README.md:73]: Link?

Comment (MichaReiser) [reply npm/rome/README.md:73]: I don't think it's worth maintaining two documentations. I recommend removing the whole section and linking to the documentation on our website.

File: npm/rome/README.md [hunk 10]
```diff
@@ -67,7 +81,9 @@ The project philosophy can be found on our [website](https://rome.tools/#philoso
 
 Contribution and development instructions can be found in [CONTRIBUTING](./CONTRIBUTING.md).
 
-Additional project coordination and real-time discussion happens on our [Discord server](https://discord.gg/rome). Remember that all activity on the Discord server is still moderated and will be strictly enforced under the project's [Code of Conduct](./CODE_OF_CONDUCT.md).
+Additional project coordination and real-time discussion happens on our [Discord server](https://discord.gg/rome). 
+
+Remember that all activity on the Discord server is still moderated and will be strictly enforced under the project's [Code of Conduct](./CODE_OF_CONDUCT.md).
```
Comment (MichaReiser) [inline npm/rome/README.md:86]:
```suggestion
Remember that all activity on the Discord server is moderated according to the project's [Code of Conduct](./CODE_OF_CONDUCT.md).
```

File: website/src/_includes/docs/getting-started.md [hunk 11]
```diff
@@ -4,89 +4,10 @@ Rome works on Linux, macOS, and Windows.
...
-  - uses: rome/setup-rome@v0.1
-    with:
-      version: latest
-  - run: rome --help
-```
-
-#### Installation on any other CI
-
-You can download and install the binary directly using `curl`:
+Open your terminal, go to your project and run:
```
Comment (MichaReiser) [inline website/src/_includes/docs/getting-started.md:7]: Can we undo these changes. I'm working on re-doing the whole getting started section

File: website/src/_includes/docs/install-editors.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
+### Install via editors
```
Comment (MichaReiser) [inline website/src/_includes/docs/install-editors.md:1]: I'll try to embed this changes in my PR but I already started rewriting the getting started section

File: website/src/_includes/docs/post-installation.md [hunk 13]
```diff
@@ -2,4 +2,6 @@
 
 Success! You're now ready to use Rome. 🥳
 
+Check how to use the [formatter](#formatter), or the how to [lint](#linter) your project.
+
```
Comment (MichaReiser) [inline website/src/_includes/docs/post-installation.md:6]: I think this is already covered by the Read the *docs*

**Review by MichaReiser — COMMENTED** [review]
We now have a bit of a problem because both of us edited the Getting started section. I'm not entirely sure how to resolve thiis.

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,77 @@
...
+### CLI
+
+- Added new command `rome version`.
+- Added new command `rome rage`.
+- Added new command `rome lsp-proxy`.
+- Added a new argument `--version`, supported by the command `rome`
+- Added new argument called `--trailing-comma`, supported by the command `rome format`.
+- Correctly show supported option for command `rome ci`, closes [#3456](https://github.com/rome/tools/issues/3456).
+- Fixed the command `rome ci` didn't run correctly under certain circumstances, closes [#3495](https://github.com/rome/tools/issues/3495). 
+- Fixed the messaging of some diagnostics, [#3460](https://github.com/rome/tools/pull/3460).
```
[the remark this answers is not in the document]
Comment (ematipico) [reply CHANGELOG.md:15]: It's a change related to the CLI itself. We use error instead of diagnostics... it's misleading

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,77 @@
...
+### Linter
+
+- **BREAKING CHANGE**: some rules have been moved to groups that fit them better, this was part of our ongoing
+  refactor of groups. Please check [#3471](https://github.com/rome/tools/pull/3471) for more details
+- Fixed issues in the `noUnreable` rule
+- Fixed false positive cases for `noNegationElse` [#3141](https://github.com/rome/tools/issues/3141)
+- Fixed false positive cases for `noUnusedVariables` [#3169](https://github.com/rome/tools/issues/3169)
+- Fixed an issue in our CFG [#3390](https://github.com/rome/tools/issues/3390)
+
+#### New rules
```
[the remark this answers is not in the document]
Comment (ematipico) [reply CHANGELOG.md:43]: I don't know to be honest. I haven't seen a linter to makes a major when introducing new rules.

File: editors/vscode/README.md (hunk 2 above)
Comment (ematipico) [reply editors/vscode/README.md:11]: I was planning to have this section in the website, but I leave that to you. So, what should I do? Add it here?

File: website/src/_includes/docs/post-installation.md (hunk 13 above)
Comment (ematipico) [reply website/src/_includes/docs/post-installation.md:6]: What do you suggest to do? To remove the paragraph?

Comment (MichaReiser) [reply website/src/_includes/docs/post-installation.md:6]: Yes

File: CHANGELOG.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (MichaReiser) [reply CHANGELOG.md:43]: Adding new rules that are recommended breaks the build of everyone using `recommended: true` because the new rules may flag new code.

[the remark this answers is not in the document]
Comment (ematipico) [reply CHANGELOG.md:43]: Exactly that. I have never seen a tool making a breaking change. No clippy, deno, eslint, etc.

[the remark this answers is not in the document]
Comment (MichaReiser) [reply CHANGELOG.md:43]: ESLint marks changes to recommended rules as breaking [source](https://github.com/eslint/eslint/commit/0fc8e62818bc8d0a0a804b59c6110818844df5f3)

grafik

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,80 @@
...
+- Added the new trailing comma option that configures where to add trailing commas. Supports the values: `all`, `es5` and `none`; refer to the [documentation](https://rome.tools/docs/#javascriptformattertrailingcomma) to learn more.
+- Improved JSX formatting [#3499](https://github.com/rome/tools/issues/3499), [#3211](https://github.com/rome/tools/issues/3211), [#3377](https://github.com/rome/tools/issues/3377)
+- Better formatting of object destructing
+- Improved formatting of test calls
+- Fixed formatting of trailing comments in arrow functions
+
+### Linter
+
+- **BREAKING CHANGE**: some rules have been moved to new groups to better reflect their purpose. This may result in Rome failing to load your configuration or suppression comments that now refer to unknown rules. Please check out [#3471](https://github.com/rome/tools/pull/3471) to learn more about the affected rules.
+- Fixed issues in the `noUnreable` rule
```
Comment (xunilrj) [inline CHANGELOG.md:38]: Typo.

File: editors/vscode/README.md [hunk 17]
```diff
@@ -2,6 +2,31 @@
 
 Adds support for the Rome language server to provide formatting, diagnostics, and code actions.
 
+## How the extension works
```
[the remark this answers is not in the document]
Comment (ematipico) [reply editors/vscode/README.md:5]: I applied this changes as part of https://github.com/rome/tools/pull/3284

File: crates/rome_cli/src/commands/help.rs [hunk 18]
```diff
@@ -18,9 +18,10 @@ const MAIN: Markup = markup! {
     - "<Emphasis>"help"</Emphasis>"         Prints this help message
 
 "<Emphasis>"OPTIONS:"</Emphasis>"
-    "<Dim>"--no-colors"</Dim>"      Disable the formatting of markup (print everything as plain text)
-    "<Dim>"--use-server"</Dim>"     Connect to a running instance of the Rome daemon server
-    "<Dim>"--version"</Dim>"        Show the Rome version information and quit
+    "<Dim>"--no-colors"</Dim>"              Disable the formatting of markup (print everything as plain text)
+    "<Dim>"--use-server"</Dim>"             Connect to a running instance of the Rome daemon server
+    "<Dim>"--version"</Dim>"                Show the Rome version information and quit
+    "<Dim>"--files-max-size"</Dim>"          The maximum allowed size for source code files in bytes. (default: 1024)
```
Comment (MichaReiser) [inline crates/rome_cli/src/commands/help.rs:24]: TODO: fix correct default

**Review by leops — APPROVED** [review]
